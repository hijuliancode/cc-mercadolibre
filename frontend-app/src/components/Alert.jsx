import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ children, className, type, width, align }) => {
  console.log('Alert.jsx');
  return (
    <div
      className={`alert alert--${type} alert--${width} alert--${align} ${className}`.trim()}
    >
      {children}
    </div>
  );
};

Alert.defaultProps = {
  type: 'primary',
  className: '',
  width: 'auto',
  align: 'left',
};

Alert.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
  className: PropTypes.string,
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'danger',
    'warning',
    'info',
    'light',
    'dark',
  ]),
  width: PropTypes.oneOf(['auto', 'full']),
  align: PropTypes.oneOf(['left', 'center', 'right']),
};

export default Alert;
