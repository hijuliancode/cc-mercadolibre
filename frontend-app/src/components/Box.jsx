import React from 'react';
import PropTypes from 'prop-types';

const Box = ({ children, className }) => (
  <div className={`box ${className}`.trim()}>{children}</div>
);

Box.defaultProps = {
  className: '',
};

Box.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
};

export default Box;
