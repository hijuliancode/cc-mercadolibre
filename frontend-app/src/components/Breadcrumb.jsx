import React from 'react';
import PropTypes from 'prop-types';

const Breadcrumb = ({ items, className }) => (
  <ul className={`breadcrumb ${className}`.trim()}>
    <li className="breadcrumb__item">{items}</li>
  </ul>
);

Breadcrumb.defaultProps = {
  className: '',
};

Breadcrumb.propTypes = {
  className: PropTypes.string,
  items: PropTypes.string.isRequired,
};

export default Breadcrumb;
