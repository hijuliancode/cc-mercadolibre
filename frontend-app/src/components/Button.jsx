/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

const Button = React.memo(({ children, className, type, styleBtn, width }) => {
  console.log('Button.jsx');
  return (
    <button
      className={`btn ${styleBtn && 'btn--in-form'} ${
        width === 'full' && 'btn--full'
      } ${className}`.trim()}
      type={type}
    >
      {children}
    </button>
  );
});

Button.defaultProps = {
  type: 'button',
  styleBtn: '',
  className: '',
  width: 'auto',
};

Button.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  styleBtn: PropTypes.oneOf(['in-form', null]),
  width: PropTypes.oneOf(['auto', 'full']),
};

export default Button;
