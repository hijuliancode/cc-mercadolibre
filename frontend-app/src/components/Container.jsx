import React from 'react';
import PropTypes from 'prop-types';

const Container = ({ children, className }) => {
  console.log('Container.jsx');
  return <div className={`container ${className}`.trim()}>{children}</div>;
};

Container.defaultProps = {
  className: '',
};

Container.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
};

export default Container;
