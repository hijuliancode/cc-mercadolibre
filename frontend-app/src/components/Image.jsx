/* eslint-disable import/no-dynamic-require */
import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ className, local, src, alt, height, width }) => {
  console.log('Image.jsx, src, alt', src, alt);
  return (
    <picture className={`image ${className}`.trim()} style={{ height, width }}>
      <img
        // eslint-disable-next-line global-require
        src={local ? require(`../images/${src}`).default : src}
        alt={alt}
        loading="lazy"
      />
    </picture>
  );
};

Image.defaultProps = {
  className: '',
  local: true,
};

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
  height: PropTypes.number.isRequired,
  local: PropTypes.bool,
  src: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
};

export default Image;
