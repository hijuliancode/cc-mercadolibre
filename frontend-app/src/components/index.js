export { default as Alert } from './Alert';
export { default as Box } from './Box';
export { default as Breadcrumb } from './Breadcrumb';
export { default as Button } from './Button';
export { default as CardResult } from './CardResult';
export { default as Container } from './Container';
export { default as Image } from './Image';
export { default as SearchBar } from './SearchBar';
export { default as Text } from './Text';
export { default as DemoComponent } from './shared/DemoComponent';
export { default as Footer } from './shared/Footer';
export { default as Header } from './shared/Header';
