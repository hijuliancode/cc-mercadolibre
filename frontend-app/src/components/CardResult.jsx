import React from 'react'
import { Link } from 'react-router-dom'
import { Image } from './Image'

export const CardResult = ({ product }) => {
  console.log('CardResult.jsx', product)
  return (
    <div className="card-result">
      <Link to={`/items/${product.id}`} title={product.title} className="card-result-image">
        <Image
          className="radius-xm"
          alt={product.title}
          height={160}
          local={false}
          src={product.thumbnail}
          width={160}
        />
      </Link>
      <div className="card-result-desc">
        <Link to={`/items/${product.id}`} title={product.title} className="card-result__title">
          <h2>{product.title}</h2>
        </Link>
        {/* <Link to={`/items/${product.id}`} title={product.seller.eshop.nick_name} className="card-result__author">
          <span>{product.seller.eshop.nick_name}</span>
        </Link> */}
        <p className="card-result__price" title={`$ ${product.price}`}>
          <span className="price-symbol">$</span>
          <span className="price-fract">{product.price}</span>
        </p>
      </div>
    </div>
  )
}
