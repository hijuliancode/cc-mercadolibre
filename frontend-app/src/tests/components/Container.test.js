import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { Container } from '../../components/Container';


describe('test in Container.jsx', () => {
  let wrapper;
  
  test('should render <Container />', () => {
    wrapper = shallow( <Container/> )
    expect( wrapper ).toMatchSnapshot()
  })
  
});
