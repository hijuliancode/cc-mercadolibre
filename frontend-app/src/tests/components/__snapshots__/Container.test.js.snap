// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in Container.jsx should render <Container /> 1`] = `
<div
  className="container"
/>
`;
